src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ProfileController.kt

```diff
@@ -23,7 +25,11 @@ class ProfileController(
 
   override fun profileV2Get(xServiceName: ServiceName): ResponseEntity<ProfileResponse> {
     val username = userService.getDeliusUserNameForRequest()
-    val userResponse = userService.getUserForProfile(username)
-    return ResponseEntity(userTransformer.transformProfileResponseToApi(username, userResponse, xServiceName), HttpStatus.OK)
+    var getUserResponse = userService.getUserForProfile(username)
+    if (getUserResponse.user != null && !getUserResponse.createdOnGet) {
+      val updateResponse = userService.updateUserFromCommunityApiById(getUserResponse.user!!.id, xServiceName)
+      getUserResponse = (updateResponse as AuthorisableActionResult.Success<GetUserResponse>).entity
+    }
+    return ResponseEntity(userTransformer.transformProfileResponseToApi(username, getUserResponse, xServiceName), HttpStatus.OK)
```

Comment:
nice, i like how this isolates the 'update on get' logic into the controller so we're not inadvertently updating elsewhere